user_guide_src/source/helpers/form_helper.rst

```diff
@@ -103,16 +103,16 @@ The following functions are available:
 
             <form action="http://example.com/index.php/email/send" class="email" id="myform" method="post" accept-charset="utf-8">
 
-        If :ref:`CSRF <cross-site-request-forgery>` filter is turned on ``form_open()`` will generate CSRF field at the beginning of the form. You can specify ID of this field by passing csrf_id as one of the ``$attribute`` array:
+        If :ref:`CSRF <cross-site-request-forgery>` filter is turned on ``form_open()`` will generate CSRF field at the beginning of the form. You can specify ID of this field by passing **csrf_id** as an element of the ``$attributes`` array:
 
         .. literalinclude:: form_helper/007.php
 
         will return::
 
             <form action="http://example.com/index.php/u/sign-up" method="post" accept-charset="utf-8">
-            <input type="hidden" id="my-id" name="csrf_field" value="964ede6e0ae8a680f7b8eab69136717d">
+            <input type="hidden" id="my-id" name="csrf_test_name" value="964ede6e0ae8a680f7b8eab69136717d">
 
-        .. note:: To use auto-generation of CSRF field, you need to turn CSRF filter on to the form page. In most cases it is requested using the ``GET`` method.
+        .. note:: To use auto-generation of CSRF field, you need to enable the :ref:`CSRF filter <enable-csrf-protection>` in **app/Config/Filters.php** file.
```

Comment:
Why do  you remove the sentence about GET request?
It is an important note.

Reply:
@kenjis It says "In most cases it is requested using the ``GET`` method." What is requested? I didn't understand. Can you please make it a bit clearer?

Reply:
To use auto-generation of CSRF field, you need to turn CSRF filter on to the form page.
In most cases the form page is requested using the ``GET`` method.
Normally, CSRF protection is required for POST/PUT/DELETE/PATCH requests, but even for GET requests, CSRF filters must be enabled for pages that display Forms.

Reply:
@kenjis Wonderful. Thanks. What about to put that explanation here in the guide?

Reply:
@kenjis To use auto-generation of the CSRF field, you need to turn on the CSRF filter. In most cases the form page is requested using the GET method. Normally, CSRF protection is required for POST/PUT/DELETE/PATCH requests, but even for GET requests, CSRF filters must be enabled for pages that display Forms.
If you enable CSRF filter with [$globals] https://codeigniter4.github.io/CodeIgniter4/incoming/filters.html#globals), it will be active for all request types. But if you enable CSRF filter with public array $methods = ['POST' => ['csrf']];, the hidden CSRF field will not be added in GET requests.

Reply:
@kenjis Expanded the note to your explanation.